### ISSUE TITLE
Dokument automatisch aufteilen (split)

### ISSUE THREAD

**Issue by djonasdev** [geimist/synOCR#34]
Gibt es derzeit eine Möglichkeit ein PDF automatisch zu zerteilen ohne ein "Split Sheet" beim Einscannen zu verwenden?

Konkret möchte ich sämtliche einseitigen Dokumente auf einmal einscannen. Diese sollen im Anschluss als separate PDFs geteilt werden.

**Comment by geimist** [comment]
Ja, dafür musst du in deinem Profil ein Kennwort definieren. Standard ist "SYNOCR-SEPARATOR-SHEET" bereits definiert. Eine Beispieldatei kannst du [HIER](https://github.com/geimist/synOCR/blob/master/SYNOCR_SEPARATOR_SHEET.pdf) laden und ausdrucken. Das kann man sich natürlich auch selbst erstellen.

**Comment by dojo90** [comment]
Danke für die schnelle Rückmeldung.

Genau das wollte ich eben nicht machen. Angenommen ich möchte ein Stapel mit 1000 einseitigen Rechnungen von 1000 unterschiedlichen Händlern einlesen, so müsste ich zwischen jede Seite ein Trennblatt legen. Wenn ich, die ich Rechnungen in die Hand nehme, weiß ich ja selbst, dass es sich hierbei um "einseitige" Dokumente handelt. Mit einem "Teile das PDF in einzelne Seiten" Profil, könnte ich diesen Fall recht einfach behandeln.

Es gibt bereits die Möglichkeit ein `postscript` (https://github.com/geimist/synOCR/wiki/03_YAML-(de)#nachstehend-eine-beschreibung-der-schl%C3%BCsselw%C3%B6rter) auszuführen. 
Was hältst du davon, wenn man bei einem Profil auch ein `pre-processing-script` oder `pre-processing-command` ausführen könnte (https://superuser.com/questions/345086/split-pdf-document-from-command-line-in-linux). Die so geteilten Dokumente würden dann alle separat eingelesen und benannt werden.

**Comment by thegodfatherrelish** [comment]
Mal eine andere Herangehensweise, hast du schon mal geschaut, ob dein Scanner diese Funktion unterstützt? Bei vielen kann man einstellen, dass von einem Dokument nur eine Seite erstellt wird, kenne ich vom Fujitsu iX1500 als auch vom Brother ADS-1700W

**Comment by geimist** [comment]
> Was hältst du davon, wenn man bei einem Profil auch ein `pre-processing-script` oder `pre-processing-command` ausführen könnte

Die Herangehensweise wäre eine andere, weil ein Postscript per Regel in der YAML-Regeldatei definiert wird. Das müsste man dann also bereit direkt im Profil definieren, weil es ja immer greifen müsste. 
Der Vorschlag von @thegodfatherrelish wäre auch meiner gewesen. Alternativ vielleicht kleines Pythonhilfsskript.



> Bei meinem **ADS-2600We** kann man das nicht einstellen

Ich bin mir ziemlich sicher, dass es diese Möglichkeit gibt, kann aber gerade nicht nachsehen.

**Comment by thegodfatherrelish** [comment]
> Ich bin mir ziemlich sicher, dass es diese Möglichkeit gibt, kann aber gerade nicht nachsehen.

Ich kann leider gerade auch nicht nachsehen, meine aber aus der Erinnerung, dass ich da noch eine oder zwei Einstellungsmöglichkeiten mehr habe beim 1700W.

**Comment by dojo90** [comment]
> > Was hältst du davon, wenn man bei einem Profil auch ein `pre-processing-script` oder `pre-processing-command` ausführen könnte
> 
> Die Herangehensweise wäre eine andere, weil ein Postscript per Regel in der YAML-Regeldatei definiert wird. Das müsste man dann also bereit direkt im Profil definieren, weil es ja immer greifen müsste. Der Vorschlag von @thegodfatherrelish wäre auch meiner gewesen. Alternativ vielleicht kleines Pythonhilfsskript.

Genau so hatte ich das auch gemeint. Vielleicht etwas missverständlich formuliert.

---

Bei beiden Drucker kann ich das nicht auswählen.. 👎

![grafik](https://github.com/geimist/synOCR/assets/17551597/77349045-3a27-4abc-82dc-cfc553ffb6ba)

![grafik](https://github.com/geimist/synOCR/assets/17551597/cdb40fe4-3a5f-4488-b831-e72ce648df9f)

**Comment by geimist** [comment]
Ich schreibe mir es mit auf, aber bitte warte nicht zeitnah auf eine Umsetzung.

**Comment by djonasdev** [comment]
Falls jemand ebenfalls auf der Suche nach dieser Lösung war.

**Comment by djonasdev** [comment]
Falls jemand ebenfalls bereits auf der Suche nach dieser Lösung ist, habe ich hier einen passenden Docker Container erstellt:
https://github.com/djonasdev/docker-inotify-pdf-tools

Die PDFs vom Scanner werden direkt in den `input` Ordner geschoben. Der `output` ist wiederum der Eingangsordner von SynOCR.
Das Aufteilen der Dokumente dauert nur wenige Sekunden.

**Comment by geimist** [comment]
In der GUI der [neusten Beta](https://geimist.eu/synOCR/synOCR_DSM7_snapshot_build.html) (1.4.99.6) gibt es jetzt die Möglichkeit, ein Dokument in einzelne Seiten aufteilen zu lassen.
